### ISSUE TITLE
Within a keyboard extension, HockeySDK  can be initialized 2x

### ISSUE THREAD

**Issue by aventurella** [bitstadium/HockeySDK-iOS#207]
The workflow is this:
- Initialize Hokey in the keyboard's `viewDidLoad`, 
- let the keyboard initialize 
- switch keyboards to any other keyboard
- switch back to the target keyboard with Hockey initialization

The following warning appears in Xcode:

```
2015-10-07 06:00:02.467 keyboard[10121:1084742] [HockeySDK] Warning: The SDK should only be initialized once! This call is ignored.
2015-10-07 06:00:02.467 keyboard[10121:1084742] [HockeySDK] Warning: startManager should only be invoked once! This call is ignored.
```

**Comment by anlinde** [comment]
Did you use the setup we document here? https://github.com/bitstadium/HockeySDK-iOS#extensions

**Comment by aventurella** [comment]
Ya, using the `HockeySDKCrashOnlyExtension`

I get
```
error: bitcode_strip /path/to/extension/keyboard/HockeySDK.framework/Versions/A/HockeySDK: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/bitcode_strip exited with 1
```

**Comment by anlinde** [comment]
Some more questions:

1. Which version of the SDK are you using?
2. Which Xcode version are you using?
3. Could you setup a sample project that reproduces both issues?

**Comment by aventurella** [comment]
SDK Version:
2015/09/18 - Stable - Version 3.8.2 - Binary Distribution 

Xcode: 
Version 7.0 (7A218)

Sample Project:
Lemme give it a whirl.

**Comment by aventurella** [comment]
Here ye be!
https://www.dropbox.com/s/3b4ayivc57ch0v1/HockeyTest.zip?dl=0

**Comment by aventurella** [comment]
It seems from here:
https://forums.developer.apple.com/thread/12443

That I might need to do this:
STRIP_BITCODE_FROM_COPIED_FILES=NO 

Trying it now.

**Comment by aventurella** [comment]
That does get me out of the bit bitcode error, but now I get a codesign error:

```
Products/Debug-iphoneos/Keyboard.appex/Frameworks/HockeySDK.framework: bundle format unrecognized, invalid, or unsuitable
Command /usr/bin/codesign failed with exit code 1
``

**Comment by anlinde** [comment]
The SDK is not a dynamic framework, so make sure not to copy it into the app bundle. Please re-read the installation instructions to make sure it is added the way it should be.

**Comment by aventurella** [comment]
The documentation: https://github.com/bitstadium/HockeySDK-iOS#setup

never discusses the use of the **CrashOnlyExtension** Framework, just the full featured framework.

And I have done the following with the **CrashOnlyExtension** framework in question:

Obviously the docs say **embeddedframework**, but I used the CrashOnlyExtension one.

```
Drag & drop HockeySDK.embeddedframework from your window in the Finder into your project in Xcode and move it to the desired location in the Project Navigator (e.g. into the group called Vendor)
A popup will appear. Select Create groups for any added folders and set the checkmark for your target. Then click Finish.
```

**Comment by anlinde** [comment]
Under section 2.4 point 3 it says:

> The SDK comes in three flavours:
> 
> i. Full featured `HockeySDK.embeddedframework`
> ii. Crash reporting only `HockeySDK.framework` in the subfolder HockeySDKCrashOnly
> iii. Crash reporting only for extensions `HockeySDK.framework` in the subfolder `HockeySDKCrashOnlyExtension` (which is required to be used for extensions when build into native frameworks).
>
> Our examples will use the full featured one.

The process will be the same for all, just use the one you want instead of the mentioned `HockeySDK.embeddedframework`

Somehow the framework is added as dynamic framework to your project, which should not happen and I have never seen this happening. That's why it tries to strip the binary and copy the framework into the app bundle and sign it, but both should never happen. I'd recommend to remove the SDK and try adding it again.

**Comment by lumaxis** [comment]
@aventurella I took a look at your sample project today. Although I couldn't reproduce your bitcode/codesign issue locally, I noticed that indeed HockeySDK.framework is in the `Copy Files` phase. This is definitely wrong and - as @anlinde mentioned - actually should never happen. Please remove it from the `Copy Files` phase in your own project and see if this solves the code signing warnings.

Also, I agree we need to improve the documentation about how to properly use the `CrashOnly` and `CrashOnlyExtension` builds and will do so soon.

About your original question: As far as I can see, our [proposed setup instructions](https://github.com/bitstadium/HockeySDK-iOS#extensions) don't work with keyboard extensions because extension's process is apparently still running after you switched to another keyboard. This means that our SDK is still running in the background when a new view controller is being initialized.

We'll try to think of another solution but this might take some time. Sorry for the inconvenience and please get back to us if you are still having other issues!

**Comment by lumaxis** [comment]
Hey @aventurella, we just released [version 3.8.3](https://github.com/bitstadium/HockeySDK-iOS/releases/tag/3.8.3) which also updates our instructions for extensions to use a static variable instead. Check it out, this should fix your log messages.
